File: app/src/main/java/mk/webfactory/template/feature/home/HomeDataSource.java
package mk.webfactory.template.feature.home;

import io.reactivex.Completable;
import io.reactivex.Maybe;

public interface HomeDataSource {

  Maybe<Object> getSomething();

  void setSomething(Object object);

  Completable deleteData();

  boolean isDataAvailable();
}

File: app/src/main/java/mk/webfactory/template/feature/home/HomeDataSource.kt
package mk.webfactory.template.feature.home

import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Maybe

interface HomeDataSource {
    fun setSomething(any: Any): Unit
    fun getSomething(): Maybe<Any>
    fun deleteData(): Completable
    val isDataAvailable: Boolean
}